go/src/staleFlagCleaner.go

```diff
@@ -0,0 +1,624 @@
+				if _, ok := sfc.valueMap[ident.Name+"."+d.Sel.Name]; ok {
+					delete(sfc.valueMap, ident.Name+"."+d.Sel.Name)
+				}
+			} else {
+				sfc.valueMap[ident.Name+"."+d.Sel.Name] = valOfExpr
+			}
+		default:
+			//Do nothing for now
+		}
+	default:
+		// other cases may come here in future
+	}
+}
+
+/*
+It will deal with *dst.ValueSpec. Values of variable stored in valueMap
+will be deleted from node.
+*/
+func (sfc *staleFlagCleaner) DelValStmt(names *[]*dst.Ident, values *[]dst.Expr) bool {
+	valOfExpr := isBot
+	delCallExpr := false
+	doNotDelLiterals := false
+	for ind := 0; ind < len((*names)); ind++ {
+		delCallExpr = false
+		doNotDelLiterals = false
+		if len((*values)) != 0 {
+			valOfExpr = sfc.evaluateExprNode((*values)[ind])
+			doNotDelLiterals = sfc.checkForBoolLiterals((*values)[ind])
+			sfc.updateValueMapPost((*names)[ind], (*values)[ind])
+		}
+
+		if (valOfExpr != isBot && !doNotDelLiterals) || (*names)[ind].Name == sfc.flagName || delCallExpr {
+			if len((*values)) != 0 {
+				var trueIdent dst.Ident
+				if valOfExpr == isTrue {
+					trueIdent.Name = strTrue
+				} else {
+					trueIdent.Name = strFalse
+				}
+				(*values)[ind] = dst.Expr(&trueIdent)
+			}
+			if len((*names)) <= 1 {
+				return true
+			}
+		}
+		valOfExpr = isBot
+	}
+	if len((*names)) == 0 {
+		return true
+	}
+	return false
+}
+
+/*
+It will deal with *dst.AssignStmt. Values of variable stored in valueMap
+will be deleted from node.
+*/
+func (sfc *staleFlagCleaner) DelAssStmt(names *[]dst.Expr, values *[]dst.Expr) bool {
+	valOfExpr := isBot
+	delCallExpr := false
+	doNotDelLiterals := false
+	for ind := 0; ind < len((*names)); ind++ {
+		//check if name is in valueName
+		valOfExpr = sfc.evaluateExprNode((*values)[ind])
+		delCallExpr = false
+
+		doNotDelLiterals = sfc.checkForBoolLiterals((*values)[ind])
+
+		// Need to see what to do with this thing
+		sfc.updateValueMapPost((*names)[ind], (*values)[ind])
+
+		if (valOfExpr != isBot && !doNotDelLiterals) || delCallExpr {
+			var trueIdent dst.Ident
+			if valOfExpr == isTrue {
+				trueIdent.Name = strTrue
+			} else {
+				trueIdent.Name = strFalse
+			}
+			(*values)[ind] = dst.Expr(&trueIdent)
+			if len((*names)) <= 1 {
+				return true
+			}
+		}
+	}
+	if len((*names)) == 0 {
+		return true
+	}
+	return false
+}
+
+// This is the post function of the dstutil.Apply
+/*
+From documentation:
+If post is not nil, and a prior call of pre didn't return false, post is
+called for each node after its children are traversed (post-order). If
+post returns false, traversal is terminated and Apply returns immediately.
+
+Working:
+This wiil be used to delete the nodes and storing the values of expressions.
+*/
+func (sfc *staleFlagCleaner) post(n *dstutil.Cursor) bool {
+	switch d := n.Node().(type) {
+	case *dst.ValueSpec:
+		//need for loop here to delete the specific varible in
+		delVal := sfc.DelValStmt(&d.Names, &d.Values)
+		if delVal {
+			n.Delete()
+		}
+	//GenDecl
+	case *dst.GenDecl:
```

Comment:
I am looking but comments explaining the intent would be really helpful indeed. Otherwise it's a back-and-forth with the (d)ast documentation to even try to figure out what's going on.